backend/tracim_backend/lib/core/workspace.py

```diff
@@ -261,6 +261,32 @@ def execute_created_workspace_actions(self, workspace: Workspace) -> None:
                 except AgendaServerConnectionError as exc:
                     logger.error(self, 'Cannot connect to agenda server')
                     logger.exception(self, exc)
+                except Exception as exc:
+                    logger.error(self, 'Something goes wrong during agenda create/update')
+                    logger.exception(self, exc)
+
+    def execute_update_workspace_actions(self, workspace: Workspace) -> None:
```

Comment:
what is the goal of this method? I believe it is to execute some actions after a workspace has been updated.

Two remarks:

- this method must not be explicitily used out of the WorkspaceApi class - this is an internal behavior
- the name is wrong.

Currently, you write:

```
workspace_api.do_some_stuff()
workspace_api.execute_update_workspace_actions()
```

This is wrong, because the user of workspace_api will probably miss to call `execute_update_workspace_actions`

The right pattern is the following: 

In client code:

```
workspace_api.do_some_stuff()
```

In WorkspaceApi code:

```
class WorkspaceApi(...):
  def do_some_stuff(self, workspace):
    """ this is the public API, which is a wrapper around private api + pre/post calls"""
    self.on_workspace_update(workspace)
    workspace = self._do_some_stuff(workspace)
    self.on_workspace_updated(workspace)
    return workspace

  def _do_some_stuff(self, workspace):
    """ this is the real API, which is a wrapped by public api. It is never called directly """
    [...]
    return workspace
```

Reply:
2 reasons.
1 - There is already a legacy "execute_created_user_actions" which work this way. I don't want to mix thoses mecanism and i do not think that the best place to refactor this mecanism. This problem as already been notify when we were talking about "plugin mecanism", it should probably been fix with it.

2 - This method should be launched after "save" method (or inside), we think about removing it but we do not did it yet. That is a huge refactor because save method is used something like 250 time, mostly in tests.

Reply:
nb: This is a drawback to apply change before commit (save do just flush), in some case it may be possible to have incoherent state because commit raise an issue but post_create/update actions has been done.

Reply:
> 1 - There is already a legacy "execute_created_user_actions" which work this way. I don't want to mix thoses mecanism and i do not think that the best place to refactor this mecanism. This problem as already been notify when we were talking about "plugin mecanism", it should probably been fix with it.

The problem is :

- you increase technical debt before the refactor (but you know this is technical debt)
- the code will be broken for all developements added before the refactoring (ie if I use workspace api in a new part of code, I have no reason "not to forgot the call to this new method)

> 2 - This method should be launched after "save" method (or inside), we think about removing it but we do not did it yet. That is a huge refactor because save method is used something like 250 time, mostly in tests.

All this explanations must appear somewhere in the code - this is not information to share in a merge request.

> For naming, i just do not change the naming from v1, that not a big deal to change them.

Reply:
I added link to new issue to fix this later https://github.com/tracim/tracim/issues/1589